### DISCUSSION TITLE
How to obtain the value of a variable at each iteration?

### DISCUSSION THREAD

**Discussion by anirudhr97** [cvxpy/cvxpy#1729] [Q&A] [answered]
I am trying to solve a SOCP problem using cvxpy. I am using the code given here - [cvxpy link](https://www.cvxpy.org/examples/basic/socp.html). I am able to get the `x` value at the end of the iterations. But, I want to access the value of the variable `x` after each iteration. Any ideas on how to do this?

**Comment by rileyjmurray** [comment, marked as answer]
Sadly, this is not possible. Your best bet would be to use CVXOPT and modify its Python implementation to support callbacks. That said, the intermediate values of "x" are not likely to be useful to you. Most optimization algorithms (essentially all which are accessible from CVXPY) generate a sequence of points that only become feasible as they approach the optimal solution. That is, most intermediate solutions produced by an algorithm violate constraints by a large margin.

**Reply by anirudhr97** [reply]
Thanks!
